### ISSUE TITLE
Avoid negative HSL angle values

### ISSUE THREAD

**Issue by jonmmease** [jackparmer/colorlover#21]
See https://github.com/plotly/plotly.py/issues/1212

Though it is technically valid (https://www.w3.org/wiki/CSS3/Color/HSL), it would be helpful if colorlover didn't produce HSL color values with negative angles (e.g. `'hsl(-5, 37%, 73%)'`), but instead wrapped these angles into the interval [0, 360] (e.g. `'hsl(355, 37%, 73%)'` in the case above).  These negative HSL values are not supported by the [tinycolor2](https://www.npmjs.com/package/tinycolor2) library that Plotly.js relies on for color handling.  See https://github.com/plotly/plotly.py/issues/1212#issuecomment-426981965 for more details.

Here's an example that produces negative hsl angle values

```python
import colorlover as cl
markercl = cl.scales['11']['qual']['Paired']
cl.interp(markercl, 16)
```
```
['hsl(76.0, 52.0%, 77.0%)',
 'hsl(157.33333333333334, 64.0%, 53.0%)',
 'hsl(162.33333333333331, 65.66666666666666%, 50.66666666666667%)',
 'hsl(90.99999999999996, 56.99999999999999%, 70.00000000000001%)',
 'hsl(107.66666666666669, 56.333333333333336%, 49.999999999999986%)',
 'hsl(124.33333333333334, 55.666666666666664%, 29.999999999999986%)',
 'hsl(-7.105427357601002e-14, 92.00000000000003%, 79.00000000000003%)',
 'hsl(-77.33333333333343, 116.00000000000003%, 105.00000000000003%)',
 'hsl(478.66666666666674, 74.66666666666666%, 38.99999999999999%)',
 'hsl(718.0, 66.0%, 19.0%)',
 'hsl(-184.3333333333336, 109.00000000000001%, 85.66666666666669%)',
 'hsl(-401.6666666666673, 121.00000000000003%, 100.33333333333337%)',
 'hsl(24.999999999999996, 103.0%, 28.99999999999998%)',
 'hsl(22.333333333333336, 105.0%, 15.0%)',
 'hsl(-59.66666666666674, -63.33333333333346%, 110.66666666666671%)',
 'hsl(-85.00000000000006, -110.00000000000009%, 128.00000000000003%)']
```

**Comment by Tunneller** [comment]
Hi, I am new to Github so no idea how to actually submit a change to Colorlover, but isnt this a one line edit to hsl = interp3( fraction, hsl_o, hsl_f ) on line #1827

Who exactly would be authorized to make that change?

Hm, I took this example https://moderndata.plot.ly/color-scales-in-ipython-notebook/

Looks like it is not quite as simple as just setting the interpretation to be modulo 360 (and the saturation modulo 100). Here is the output that I get

https://plot.ly/~Tunneller/4

However, at least now the example makes a plot without complaining...

**Comment by jonmmease** [comment]
Closed by https://github.com/jackparmer/colorlover/pull/25 and will be released on version 0.3

```python
import colorlover as cl
markercl = cl.scales['11']['qual']['Paired']
cl.interp(markercl, 16)
```
```
['hsl(200.0, 52.0%, 77.0%)',
 'hsl(202.66666666666666, 64.0%, 53.0%)',
 'hsl(166.33333333333334, 65.66666666666667%, 50.666666666666664%)',
 'hsl(91.0, 57.0%, 70.0%)',
 'hsl(107.66666666666666, 56.333333333333336%, 50.0%)',
 'hsl(77.33333333333331, 68.0%, 53.00000000000001%)',
 'hsl(0.0, 92.0%, 79.0%)',
 'hsl(359.3333333333333, 83.33333333333333%, 58.99999999999999%)',
 'hsl(10.333333333333314, 85.0%, 56.33333333333333%)',
 'hsl(33.0, 97.0%, 71.0%)',
 'hsl(30.333333333333332, 99.0%, 56.99999999999999%)',
 'hsl(352.6666666666667, 76.66666666666669%, 58.66666666666666%)',
 'hsl(280.0, 30.0%, 76.0%)',
 'hsl(272.6666666666667, 38.66666666666666%, 53.33333333333336%)',
 'hsl(319.3333333333334, 62.000000000000036%, 54.666666666666686%)',
 'hsl(60.0, 100.0%, 80.0%)']
```

```python
HTML(cl.to_html(markercl))
```
![screen shot 2019-01-19 at 10 43 03 am](https://user-images.githubusercontent.com/15064365/51428871-0a864100-1bd7-11e9-9281-58882f12d195.png)


```python
HTML(cl.to_html(cl.interp(markercl, 16)))
```
![screen shot 2019-01-19 at 10 43 11 am](https://user-images.githubusercontent.com/15064365/51428875-0e19c800-1bd7-11e9-9d18-b6eabad57904.png)
